libflux/flux-core/src/semantic/nodes.rs

```diff
@@ -1641,21 +1822,24 @@ pub struct IdentifierExpr {
 
 impl IdentifierExpr {
     fn infer(&mut self, infer: &mut InferState<'_>) -> Result {
-        let poly = infer.env.lookup(&self.name).ok_or_else(|| {
-            located(
-                self.loc.clone(),
-                ErrorKind::UndefinedIdentifier(self.name.to_string()),
-            )
-        })?;
-
-        let (t, cons) = infer::instantiate(poly.clone(), infer.sub, self.loc.clone());
+        let (t, cons) = infer.lookup(&self.loc, &self.name)?;
         self.typ = t;
         Ok(cons)
     }
     fn apply(mut self, sub: &Substitution) -> Self {
         self.typ = self.typ.apply(sub);
         self
     }
+
+    fn vectorize(&self, env: &VectorizeEnv) -> Result<Self> {
```

Comment:
I am not clear on two functions using the same function signature. How does this work in rust?
Also, there are 4 vectorize methods in this file, I can see they are for different purposes here. but if possible could you please rename it to a more specific task each of the methods is doing. something like vectorize_pkg

Reply:
All of the `vectorize` methods are on different types, so you have `IdentifierExpr:;vectorize`, `Expression::vectorize` etc. I might have done `VectorizeEnv::vectorize_pkg`, `VectorizeEnv::vectorize_identifier` etc instead but `fn infer` and `fn apply` already exist in the same manner as the added `vectorize` methods so I stuck with the same naming scheme.

Reply:
That's alright then.